Lesson3

VeronikaMads opened the pull request:
1. решение к задачам 1 и 2.

VeronikaMarss committed:
ДЗ по третьему уроку /1
1. решение к задачам 1 и 2.

pavelTMS commented on Lesson3/src/HomeWork2.java:
```diff
@@ -0,0 +1,20 @@
+        int numder = scanner.nextInt();
+        if (numder > 0) {
+            ++numder;
+        }
```
else добавить надо. если число больше 0 то в остальные иф нам не нужно заходить и проверять условия.

pavelTMS commented on Lesson3/src/HomeWork.java:
```diff
@@ -0,0 +1,19 @@
+        for (int i = 0; i < lenght; i++) {
+            mass[i] = scanner.nextInt();
+        }
+        System.out.print("Введите значения массива: ");
```
тут вы имели ввиду "Вывод значений массива" ?

pavelTMS commented:
посмотрел!

VeronikaMarss committed:
ДЗ по третьему уроку /2
1. В первой задаче исправила ошибку в sout.
2. Во второй задаче добавила Else. (Хотела узнать в чем разница. в чем минус того кода который был изначально, без else).
(остальные задачи будут после лекции, или вместе с ДЗ по 4 лекции)

VeronikaMarss committed:
ДД! Я пытаюсь разобраться как делать все дз в одном файле. и задачу с программистами никак не могу понять.
из вашего файла..
1. Строчка 3   countDevs(103); - вывызвет метод " public static void countDevs(int count)"? Это озночает что int count = 103; и именно это число мы должны вевести (103 программиста)?  я правильно понимаю ?
public class HomeWork3 {
    public static void main(String[] args) {
        countDevs(103);}
        public static void countDevs(int count) {
                int....}
                }

2.

VeronikaMarss committed:
ДЗ

VeronikaMarss committed:
ДЗ

VeronikaMarss committed:
ДЗ

VeronikaMarss committed:
ДЗ

VeronikaMarss committed:
ДЗ

pavelTMS commented on Lesson3/src/HomeWork3.java:
```diff
@@ -0,0 +1,24 @@
+public class HomeWork3 {
+    public static void main(String[] args) {
+        countDevs(i);
```
что произойдет в 3 строчке. это вызов метода countDevs(), в метод countDevs передаем переменную i.
между 2 и 3 строчкой нужно еще объявить переменую i  и присвоить ей значение например 103
`int i = 103;`

pavelTMS replied:
т.е передали число 103 в метод countDevs который выполнит логику по правильному склонению слова программист.
и в итоге напечатает на консоль 103 программиста.

pavelTMS commented on Lesson3/src/HomeWork3.java:
```diff
@@ -0,0 +1,24 @@
+     */
+    public static void countDevs(int count) {
+        int p = count % 100;
+        int p2 = p / 10;
```
тут надо `int p2 = p % 10;`

pavelTMS commented on Lesson3/src/HomeWork3.java:
```diff
@@ -0,0 +1,24 @@
+    public static void countDevs(int count) {
+        int p = count % 100;
+        int p2 = p / 10;
+        if (p2 == 1){
```
осталось другие условия написать.

pavelTMS reviewed (commented):
посмотрел!

VeronikaMarss committed:
ДЗ2

VeronikaMarss committed:
ДЗ2

VeronikaMarss committed:
ДЗ2

VeronikaMarss committed:
ДЗ2

VeronikaMarss committed:
ДЗ к третьему уроку.
1. Задача 1 не выводится, пишет не хвататет еще элемента.
2. В задаче не поняла как сделать остальные треугольники.
3.В задаче 5 не поняла как применять  Arrays.sort (на лекции наказывали заданный массив, с тремя числами)
4. 6  задача тоже не доделана.

VeronikaMarss committed:
Удаляю iml файл
